### PR TITLE
readonly attribute of GroupedList field throws a warning

### PR DESCRIPTION
**Pull request by colivier**
Steps to reproduce the issue

1) Set error reporting in Joomla configuration to maximum.
2) Use a form with a GroupedList field.
3) Select some items and save the form
4) Put the attribute readonly of the field by code or by the manifest of the form.
5) Reopen the form
6) You'll see a PHP warning htmlspecialchars() expects parameter 1 to be string, array given in /libraries/joomla/form/fields/groupedlist.php on line 169

Suggested solution
I suggest to do the same thing than in the List field.

### CODE REVIEW

**Comment by bertmert** [comment]
Even if Icouldn't see a warning (error_reporting Maximum), you're right, this form field doesn't support readonly.

@test Success after patch. Tested disabled, readonly, multiple.

Because it was hard for me to find a form with groupedlist field:

My test:
Add to /administrator/.../com_content/...forms/article.xml
below line 147 (< fieldset name="basic" label="COM_CONTENT_ATTRIBS_FIELDSET_LABEL" >)

these lines:

<field name="mylistvalue" 
 type="groupedlist" 
 default="" 
 label="Select from groupedlist" 
 description="">
<group label="Group 1">
<option value="0">Option 1</option>
<option value="1">Option 2</option>
</group>
<group label="Group 2">
<option value="3">Option 3</option>
<option value="4">Option 4</option>
</group>
<option value="5">Option 5</option>
<option value="6">Option 6</option>
</field>

Now you see this field in article edit view > Tab Options.
And can add readonly and/or multiple and so on....

**Comment by colivier** [comment]
bertmert, thank you very much for the test

**Comment by zero-24** [comment]
@test successful thanks @colivier and @bertmert Moving to RTC now!<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6494">issues.joomla.org/joomla-cms/6494</a>.</sub>

File: libraries/joomla/form/fields/groupedlist.php [hunk 1]
```diff
@@ -144,12 +144,17 @@ protected function getInput()
...
 		// Initialize some field attributes.
 		$attr .= !empty($this->class) ? ' class="' . $this->class . '"' : '';
-		$attr .= $this->disabled ? ' disabled' : '';
 		$attr .= !empty($this->size) ? ' size="' . $this->size . '"' : '';
 		$attr .= $this->multiple ? ' multiple' : '';
 		$attr .= $this->required ? ' required aria-required="true"' : '';
 		$attr .= $this->autofocus ? ' autofocus' : '';
 
+		// To avoid user's confusion, readonly="true" should imply disabled="true".
+		if ($this->readonly || $this->disabled)
```
Comment (phproberto) [inline libraries/joomla/form/fields/groupedlist.php]: `readonly` & `disabled` are different things. A `readonly` field will send the field in the form but a `disabled`. So it's not a good idea to do this.

**Comment by colivier** [comment]
The desabled attribute is set when the readonly attribute is present to avoid user's confusion, otherwise the grouplist could be changed by the user.

In addition, when the readonly  attribute is present, the name of the groupedlist is not set and in the same time another input of type hidden is created with the name of the grouplist to send the value of the field.

**Comment by Kubik-Rubik** [comment]
What is the state of this PR? @colivier @wilsonge @phproberto

**Comment by colivier** [comment]
The tests were successful but now I do not know what to do

**Comment by roland-d** [comment]
Setting this back to Pending because @phproberto made a very good point with his comment on the code:
https://github.com/joomla/joomla-cms/pull/6494/files#diff-b1b272be917894e1db8f736bbc4191a0R153

I read the reply by @colivier but I am not sure I fully understand. If in the XML definition the field is set to be readonly, the field shouldn't be disabled, it should be made readonly. Sure a field can be changed by the user but that is up to the person who decides to use this field with such a setting.

So I do think we need to change the readonly to be readonly and not disabled.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="http://issues.joomla.org/tracker/joomla-cms/6494">issues.joomla.org/joomla-cms/6494</a>.</sub>

**Comment by colivier** [comment]
Are you test the steps to reproduce the issue (see above) ?

**Comment by roland-d** [comment]
@colivier Yes, I was able to reproduce the issue. Your change fixes the issue but also touches another issue. That issue is that the readonly and disabled are treated the same way. That is what Roberto commmented on. However, seeing that the same "fix" is applied in the list field, I would say let's accept your proposal.

**Pull request merged**